# IBDHap-detector
 
IBDHap-detector is pipeline of detecting IBD or Haplotype segements based on phasing genotypes.

# Installation
Requires HMM, pbapply, parallel, randomForest, MASS and smotefamily packages in R

# Running
These scripts could be used to reproduce the results in our papers. If you want to use them to analyze your own data, you may need to transfer your data to the suitable format for this pipeline.

## data preparation
 You need to prepare correct.haps.filtered.formated.txt, formated_w221_snp.filtered.txt and genetic maps first.
 
 correct.haps.filtered.formated.txt gives the following format:
 ```
 m_chr1_40	1_57438086	G	A	1	74423
 m_chr1_40	1_57438105	G	A	1	74442
 m_chr1_40	1_57438151	C	T	1	74488
 m_chr1_40	1_57438276	C	A	1	74613
 ```
 The first column is bin id, the second column show the chromosome and position of the SNP in DASZ genome, which are seprated by the underline. The third and fourth columns indicate the two phased SNPs. The last two columns are chromosome and position of the SNP in Fudingdabai genome.
 
 formated_w221_snp.filtered.txt gives the following format:
 ```
 X3W1.1	X3W1.10	X3W1.12	X3W1.13	X3W1.14	X3W1.15	X3W1.16
 1_107784	AG	A	A	A	A	NA	A
 1_108003	GA	G	G	G	G	GA	G
 ```
 Rownames indicate the chromosome and position of the SNP in DASZ genome, which is same as the second column the  correct.haps.filtered.formated.txt. Colnames show the sample IDs. Missing genotypes are showing as NA.
 
 The example of genetic maps are in the directory "linkage_group", which is generated by MSTmap software.
 
 
